lib/appnexusapi/service.rb

```diff
@@ -54,6 +55,19 @@ def get(params={})
     end
   end
 
+  def get_all(params = {})
+    responses = []
+    last_responses = get(params)
+
+    while last_responses.size > 0
+      responses += last_responses
+      last_responses = get(params.merge('start_element' => responses.size))
+      sleep(1) # The gem has no error handling at all for rate limit errors; sleeping for a second prevents errors
```

Comment:
the rate limit is 100 per minute
```
{"response"=>
  {"error_id"=>"SYSTEM",
   "error"=>
    "You have exceeded your request limit of 100 per 60 seconds for this Default, please wait and try again or contact AppNexus for higher limits",
```
so sleeping for 0.6 seconds would be living on the edge, but realistically this shouldn't work like this at all - it should just catch rate limit exceptions and then (and only then) sleep/wait/retry,.

given that it can't really be close to what it should be, i figure 1s is fine.  guarantees no issues